File: kt/2274.kt
class Solution {
    fun maxConsecutive(bottom: Int, top: Int, special: IntArray): Int {
        // add the bottom and top to the special array and sort it
        val specialList = special.toMutableList()
        specialList.add(bottom - 1)
        specialList.add(top + 1)
        specialList.sort()
        return specialList.zipWithNext { a, b -> b - a - 1 }.max()
    }
}

File: py/2274.py
class Solution:
    def maxConsecutive(self, bottom: int, top: int, special: List[int]) -> int:
        floors = sorted([bottom - 1, top + 1] + special)
        return max(floors[i+1] - floors[i] - 1 for i in range(0, len(floors) - 1))